semgrep/semgrep/target_manager.py

```diff
@@ -391,9 +602,20 @@ def filtered_files(self, lang: Language) -> FrozenSet[Path]:
         explicit_files, _ = partition_set(lambda p: p.is_file(), files)
 
         targets = self.expand_targets(directories, lang, self.respect_git_ignore)
-        targets = self.filter_includes(targets, self.includes)
-        targets = self.filter_excludes(targets, [*self.excludes, ".git"])
-        targets = self.filter_by_size(targets, self.max_target_bytes)
+        targets = self.filter_includes(
```

Comment:
I wonder if it might be less cognitive load to instead of having the decorator and specifying a removal_log, just do it inline? 
i.e.
```
targets, filtered_out = self.filter_includes(self.includes, targets)
log_removed_paths(filtered_out, CLI_INCLUDES)
```

Much more obvious than having to know to pass a removal_log into a decorator that will silently do nothing if removal_log is not specified

Reply:
My thoughts that led to deciding against that were: it would more than double the amount of symbols copy-pasted here (and the other method that does filtering as well), so updating this would get a lot more error-prone

Reply:
Basically trying to mitigate the last line effect https://pvs-studio.com/en/blog/posts/cpp/0260/

Reply:
IMO that's fine and those errors should be caught in tests anyway. Trying to decrease the complexity in reading the code. With the decorator someone has to go to the function declaration of filter_includes and the function declaration of the decorator. With the suggested approach it is clear even without loading that state that we are filtering out things and logging. the removed things.

Reply:
re optimizing for reading: https://www.python.org/dev/peps/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds

and zen of python `Explicit is better than implicit`

Reply:
Alternatively we can modify the filter_includes and filter_excludes to always perform the log behavior and change the function name to reflect that.